src/app/pages/ProposalDetailsPage/hooks.ts

```diff
@@ -116,28 +117,54 @@ export const useVoteStats = (network: Network, proposalId: number): VoteStats =>
   }
 }
 
-export const useWantedVoteType = () =>
-  useTypedSearchParam<VoteType>('vote', 'any', {
+export const useVoteFiltering = (t: TFunction) => {
+  const [wantedType, setWantedType] = useTypedSearchParam<VoteType>('vote', 'any', {
     deleteParams: ['page'],
   })
+  const [wantedNameInput, setWantedNameInput] = useTypedSearchParam('voter', '', { deleteParams: ['page'] })
+  const wantedNamePattern = wantedNameInput.length < 3 ? undefined : wantedNameInput
+  const nameError = !!wantedNameInput && !wantedNamePattern ? t('tableSearch.error.tooShort') : undefined
+  const hasFilters = wantedType !== 'any' || !!wantedNamePattern
+  const clearFilters = () => {
+    setWantedType('any', { deleteParams: ['voter', 'page'] })
```

Comment:
This is a bit unintuitive: why is `vote` missing in `deleteParams: ['voter', 'page']`? It's because `setWantedType` sets `vote` to the default value and that deletes it.

I think I would choose:
```ts
  setWantedType('any')
  setWantedNameInput('')

// or

  setSearchParams((searchParams) => {
    searchParams.delete('vote')
    searchParams.delete('voter')
    searchParams.delete('page')
    return searchParams
  })
```

Reply:
Unexpectedly `setWantedType('assignsFirst-deletesSecond', { deleteParams: ['vote', 'voter', 'page'] })` would delete it too

Reply:
> This is a bit unintuitive: why is `vote` missing in `deleteParams: ['voter', 'page']`? It's because `setWantedType` sets `vote` to the default value and that deletes it.

Exactly.

> 
> I think I would choose:

> ```ts
>   setWantedType('any')
>   setWantedNameInput('')

Yes, that is the natural choice, but unfortunately it doesn't work. I think the issue is with react router's implementation. Setting multiple search query parameter in quick succession causes race conditions.

 
> // or
> 
>   setSearchParams((searchParams) => {
>     searchParams.delete('vote')
>     searchParams.delete('voter')
>     searchParams.delete('page')
>     return searchParams
>   })
> ```

Yes, that would work. The problem is that if we are going that way, we can't reap any of the benefits of the already defined hook for working with the vote type.  By going around the existing hook, we are hurting encapsulation a little bit.

OTOH we are hurting it already (by manually deleting `page` and `voter`, so it doesn't really matter. I will go with this versino.

> Unexpectedly `setWantedType('assignsFirst-deletesSecond', { deleteParams: ['vote', 'voter', 'page'] })` would delete it too

Yes that works too, but I don't think it's any more intuitive to read.

Reply:
Updated accordingly